src/System.IO.Pipelines/ref/System.IO.Pipelines.cs

```diff
@@ -75,7 +75,7 @@ public abstract partial class PipeWriter : System.Buffers.IBufferWriter<byte>
         public abstract System.Threading.Tasks.ValueTask<System.IO.Pipelines.FlushResult> FlushAsync(System.Threading.CancellationToken cancellationToken = default(System.Threading.CancellationToken));
         public abstract System.Memory<byte> GetMemory(int sizeHint = 0);
         public abstract System.Span<byte> GetSpan(int sizeHint = 0);
-        public abstract void OnReaderCompleted(System.Action<System.Exception, object> callback, object state);
+        public virtual void OnReaderCompleted(System.Action<System.Exception, object> callback, object state) { }
```

Comment:
```suggestion
        [Obsolete("OnReaderCompleted may not be invoked on all implementations of PipeWriter. This will be removed in a future release.")]
        public virtual void OnReaderCompleted(System.Action<System.Exception, object> callback, object state) { }
```